pimd/pim_pim.c

```diff
@@ -661,23 +711,6 @@ int pim_msg_send(int fd, pim_addr src, pim_addr dst, uint8_t *pim_msg,
 	to.sin_family = AF_INET;
 	to.sin_addr = dst;
 	tolen = sizeof(to);
-#else
-	struct ip6_hdr *ip = (struct ip6_hdr *)buffer;
-	struct sockaddr_in6 to = {};
-	int sendlen = sizeof(*ip) + pim_msg_size;
-
-	ip->ip6_flow = 0;
-	ip->ip6_vfc = (6 << 4) | (IPTOS_PREC_INTERNETCONTROL >> 4);
-	ip->ip6_plen = htons(pim_msg_size);
-	ip->ip6_nxt = PIM_IP_PROTO_PIM;
-	ip->ip6_hlim = ttl;
```

Comment:
fields ttl, msg_start and tolen is no more used now in case of PIMv6 compilation. It will give throw unused error. Please move those variables under PIM_IPV4.

Reply:
Fixed

Reply:
So now the TTL will be filled by the kernel?

Reply:
So the kernel will fill some default TTL values but for some of the PIM packets the TTL should be set to 1, right?

Reply:
Yeah the hello packets are set to hop limit 1 by kernel which I checked during testing